src/frontends/tensorflow_common/src/op/ones_like.cpp

```diff
@@ -19,27 +19,56 @@ namespace tensorflow {
-    return {ones_like};
-}
+    auto complex_type_mark_x = as_type_ptr<ComplexTypeMark>(x.get_node_shared_ptr());
+    if (complex_type_mark_x) {
+        x = complex_type_mark_x->input_value(0);
+        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto minus_one = make_shared<v0::Constant>(element::i32, Shape{1}, -1);
+        auto x_real = make_shared<v8::Gather>(x, gather_index_real, minus_one)->output(0);
+
+        Output<Node> shape_of_real = make_shared<v3::ShapeOf>(x_real, element::i32);
+        auto one_const = create_same_type_const_scalar<int32_t>(x_real, 1);
+        auto one_dim = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+        shape_of_real = make_shared<v0::Concat>(OutputVector{one_dim, shape_of_real}, 0);
+        // create a tensor of zeros of shape with extra dimension
+        Output<Node> ones_like_one = make_shared<v3::Broadcast>(one_const, shape_of_real);
+        // remove extra dimension by squeezing
+        auto zero_dim_ind = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+        ones_like_one = make_shared<v0::Squeeze>(ones_like_one, zero_dim_ind);
+
+        // For zeroes:
+        auto zero_const = create_same_type_const_scalar<int32_t>(x_real, 0);
+        // create a tensor of zeros of shape with extra dimension
+        Output<Node> ones_like_zero = make_shared<v3::Broadcast>(zero_const, shape_of_real);
+        // remove extra dimension by squeezing
+        ones_like_zero = make_shared<v0::Squeeze>(ones_like_zero, zero_dim_ind);
 
+        auto ones_like_output = make_shared<v0::Concat>(OutputVector{ones_like_one, ones_like_zero}, -1);
+        auto ones_like_output_complex =
+            make_shared<ComplexTypeMark>(ones_like_output, ones_like_complex->get_complex_part_type())
+                set_node_name(node.get_name(), ones_like_output.get_node_shared_ptr());
+        return {ones_like_output_complex};
+    } else {
+        Output<Node> shape_of = make_shared<v3::ShapeOf>(x, element::i32);
+        auto one_const = create_same_type_const_scalar<int32_t>(x, 1);
+
+        // in case of x to be scalar, we need handle it more specifically
+        // since Broadcast supports only broadcasting to rank greater 0
+        // we have to introduce extra dimension for input scalar case
+        auto one_dim = make_shared<v0::Constant>(element::i32, Shape{1}, 1);
+        shape_of = make_shared<v0::Concat>(OutputVector{one_dim, shape_of}, 0);
+
+        // create a tensor of zeros of shape with extra dimension
+        Output<Node> ones_like = make_shared<v3::Broadcast>(one_const, shape_of);
+        // remove extra dimension by squeezing
+        auto zero_dim_ind = make_shared<v0::Constant>(element::i32, Shape{1}, 0);
+        ones_like = make_shared<v0::Squeeze>(ones_like, zero_dim_ind);
+        set_node_name(node.get_name(), ones_like.get_node_shared_ptr());
+        return {ones_like};
+    }
```

Comment:
please remove `else` branch and put code outside as it was. It will be equivalent and simpler to review non-complex type case to make sure no changes for non-complex type:)